matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/identity/DefaultIdentityService.kt

```diff
@@ -136,67 +130,64 @@ internal class DefaultIdentityService @Inject constructor(
         return identityStore.getIdentityData()?.identityServerUrl
     }
 
-    override fun startBindThreePid(threePid: ThreePid, callback: MatrixCallback<Unit>): Cancelable {
+    override suspend fun startBindThreePid(threePid: ThreePid) {
         if (homeServerCapabilitiesService.getHomeServerCapabilities().lastVersionIdentityServerSupported.not()) {
-            callback.onFailure(IdentityServiceError.OutdatedHomeServer)
-            return NoOpCancellable
+            throw IdentityServiceError.OutdatedHomeServer
         }
 
-        return taskExecutor.executorScope.launchToCallback(coroutineDispatchers.main, callback) {
+        withContext(coroutineDispatchers.main) {
```

Comment:
I'm wondering if this is necessary to keep the `withContext()` call here (and for other methods too). WDYT?

Reply:
Yeah I was thinking about this. We can (and should) drop the `withContext` here.
To keep things simple I left it in, to mostly preserve previous behaviour.
Once all the callbacks have been converted and everything is consistently marked as `suspend`, I was planning to review all the uses of `withContext` and remove the unnecessary ones.
Although....... most of these have been pretty simple so far but I only realized that after like the 9th PR, by which point it was too late 😂.
We can just remove them; I'm at work right now so I can do so in the evening.

Reply:
Removed